src/App/Pages/TabsPage.cs

```diff
@@ -70,6 +72,15 @@ public TabsPage(AppOptions appOptions = null, PreviousPageInfo previousPage = nu
             {
                 ResetToSendPage();
             }
+            
+        }
+
+        protected override async void OnAppearing()
+        {
+            if (await _keyConnectorService.UserNeedsMigration())
```

Comment:
I looked into this pretty extensively today. I thought we could move the page to `App.xaml.cs` and use the MessagingService to bring it up after a sync. The problem is that we don't sync when we land in the vault. We sync when we authenticate with SSO or login, but if the vault is just locked there is no sync. 

I also considered saving "needs migration" to storage on the sync, then checking later and calling `RemoveMasterPasswordPage`. This seemed just a roundabout way of calling the page though and I didn't see an advantage. We would still have to attach it to a page somewhere.

I ended up just attaching it to the `LockPage`. 

Let me know what you think @eliykat, here's the commit:
af6525dcafb04eaea763d437593826ed1f9f9235

Reply:
> The problem is that we don't sync when we land in the vault. We sync when we authenticate with SSO or login, but if the vault is just locked there is no sync.

But the client won't know if the user uses key connector _until_ there's a sync. A sync is the earliest possible time that we can know key connector is involved and redirect to the remove-password page. Your first suggestion of moving it to `App.xaml.cs` and using the messagingService seems fine to me if that was your only concern.

Reply:
I see your point. I believe we'll still have to attach the redirect to a page though. The initial sync happens before we reach the vault, while the SSO, lock, and login pages are showing. We can avoid showing it on those pages but unless the user initiates a sync manually, they won't be prompted with the remove-password page.

Reply:
`TabsPage` is the common landing page when you login or unlock, right? That seems reasonable, unless @Hinton has a better idea.

Reply:
I think we need to check with @mportune-bw, I don't really have a better idea.

Reply:
The core issue is that sync in mobile is performed asynchronously so the app is still usable while the vault is updated.  Unfortunately that's becoming problematic as more non-vault data is added to the sync model.

@vincentsalucci and I recently ran into this with the [Reset Master Password](https://github.com/bitwarden/mobile/pull/1492) flow, and eventually added the flag to the auth response (where it belongs imo).  Otherwise the experience becomes a clumsy mess depending on the size of the vault and network conditions (i.e. random screens popping up while the user is already logged in, and no guarantee the app will even be in the foreground when it happens, etc.)

@cscharf @kspearrin I believe a larger discussion is warranted as the "put everything in sync" approach is forcing mobile to make bad design decisions.

Reply:
@jlf0dev Sounds like this prompt isn't time-sensitive so using `OnAppearing` on the TabsPage is fine for now.

Reply:
I've moved the call for RemoveMasterPasswordPage to the `OnAppearing` of the TabsPage

ee02697

Reply:
So now we do the check on `TabsPage.OnAppearing` and send the message to `App.xaml.cs`, which then creates the new page. Seems fine